pkg/provider/handlers/functions.go

```diff
@@ -181,3 +180,19 @@ func buildLabelsAndAnnotations(ctrLabels map[string]string) (map[string]string,
 
 	return labels, annotations
 }
+
+func ListNamespaces(client *containerd.Client) []string {
+	set := []string{}
+	namespaces, err := client.NamespaceService().List(context.Background())
```

Comment:
It seems like we can use labels like we do with Kubernetes:

https://github.com/containerd/containerd/blob/main/docs/namespaces.md

Reply:
Sure.

We may to do some changes in deployment code. As we don't provide a way to create namespace from faas-cli in containerd. We would have to apply label on namespace while creating any function.

Based on this there could be some changes required in each method.

**Deploy Function**
1. Check if namespace exist and  label is applied on it. If Yes, then deploy function. If No, Create namespace and apply label on it.
2. If namespace already exist and no label is applied. Send error to user.

**Delete/invokde/List/Update Function**
1. Check if label exist. If Yes, perform operation. If No, throw error.


Please do let me know, if there is something I missed.

Reply:
@alexellis What do you think about above suggestion or do you think user should use Containerd Client CLI to create and manage namespace?

Reply:
The default namespace should be as defined, but any additional namespaces must be labelled using the CLI. This would match how the Kubernetes support works.

Reply:
Changes are completed.